src/edgekv/ekv-service.ts

```diff
@@ -10,121 +10,336 @@ const INIT_EKV_TIMEOUT = 120000;
+    name: tokenName,
+    allowOnProduction: allowOnProd,
+    allowOnStaging: allowOnStg,
+    restrictToEwids: ewids,
+    expiry: expiry,
+    namespacePermissions: permissionList,
   };
-  return httpEdge.postJson(`${EDGEKV_API_BASE}/tokens`, body, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT), ekvMetrics.createToken).then(r => r.body).catch(err => error.handleError(err));
+  return httpEdge
+    .postJson(
+      `${EDGEKV_API_BASE}/tokens`,
+      body,
+      cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT),
+      ekvMetrics.createToken
+    )
+    .then((r) => r.body)
+    .catch((err) => error.handleError(err));
 }
 
 export function getSingleToken(tokenName: string) {
-  return httpEdge.getJson(`${EDGEKV_API_BASE}/tokens/${tokenName}`, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT), ekvMetrics.readToken).then(r => r.body).catch(err => error.handleError(err));
+  return httpEdge
+    .getJson(
+      `${EDGEKV_API_BASE}/tokens/${tokenName}`,
+      cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT),
+      ekvMetrics.readToken
+    )
+    .then((r) => r.body)
+    .catch((err) => error.handleError(err));
 }
 
 export function getTokenList(incExpired: boolean) {
-  var qs: string = "";
+  var qs: string = '';
   if (incExpired) {
-    qs += `?includeExpired=${incExpired}`
+    qs += `?includeExpired=${incExpired}`;
   }
-  return httpEdge.getJson(`${EDGEKV_API_BASE}/tokens${qs}`, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT), ekvMetrics.readTokenList).then(r => r.body).catch(err => error.handleError(err));
+  return httpEdge
+    .getJson(
+      `${EDGEKV_API_BASE}/tokens${qs}`,
+      cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT),
+      ekvMetrics.readTokenList
+    )
+    .then((r) => r.body)
+    .catch((err) => error.handleError(err));
 }
 
 export function revokeToken(tokenName: string) {
-  return httpEdge.deleteReq(`${EDGEKV_API_BASE}/tokens/${tokenName}`, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT), ekvMetrics.deleteToken).then(r => r.body).catch(err => error.handleError(err));
+  return httpEdge
+    .deleteReq(
+      `${EDGEKV_API_BASE}/tokens/${tokenName}`,
+      cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT),
+      ekvMetrics.deleteToken
+    )
+    .then((r) => r.body)
+    .catch((err) => error.handleError(err));
 }
 
 export function modifyAuthGroupPermission(namespace: string, groupId: number) {
-  let body = { "groupId": groupId };
-  return httpEdge.putJson(`${EDGEKV_API_BASE}/auth/namespaces/${namespace}`, body, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT), ekvMetrics.modifyAuthGroup).then(r => r.body).catch(err => error.handleError(err));
+  let body = { groupId: groupId };
+  return httpEdge
+    .putJson(
+      `${EDGEKV_API_BASE}/auth/namespaces/${namespace}`,
+      body,
+      cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT),
+      ekvMetrics.modifyAuthGroup
+    )
+    .then((r) => r.body)
+    .catch((err) => error.handleError(err));
 }
 
+// TODO: Should `groupId` parameter be optional?
 export function listAuthGroups(groupId: number) {
```

Comment:
Hi @bmatthew, I wonder if this `groupId` parameter should be optional.

Reply:
Yeah it should be. The call into here in listAuthGroups in ekv-handler (line 349) seems like it works around the optionality missing by calling with a variable assigned to null so if you change this you could modify that as well to just call with no variable

Reply:
Do we want to handle some invalid cases like passing non-positive numbers?

Reply:
Invalid options should already be handled by the OPEN API called here so it shouldn't be needed.